pllug-presentation-system/presentation-data/concretepresentationelementfactory.cpp

```diff
@@ -0,0 +1,29 @@
+#include "concretepresentationelementfactory.h"
+
+#include <QString>
+
+#include "presentationelement.h"
+#include "header.h"
+#include "paragraph.h"
+#include "separator.h"
+std::unique_ptr<PresentationElement> ConcretePresentationElementFactory::create(const QString &html) const
+{
+    int position = html.indexOf(">");
+    QString tag = html.left(position + 1);
+    if (tag[1] == 'h')
+    {
+        return std::unique_ptr<Header>(new Header(html));
```

Comment:
Please, use std::make_unique<Header>(html) instead of  std::unique_ptr<Header>(new Header(html))